### PR TITLE
fix(common): add portalhook failsafe

### PR DESCRIPTION
**Pull request by stavros-k**
**Description**
There are some edgecases where the portalhookconfigmap isn't detected correctly.
We should take that into account.

Also uses get instead of index to fetch the portalhook value and adds a failback to try-and-fetch the websecure portalhook if failed to fetch the given entrypoint.

**⚙️ Type of change**

- [ ] ⚙️ Feature/App addition
- [x] 🪛 Bugfix
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**

**📃 Notes:**

**✔️ Checklist:**

- [ ] ⚖️ My code follows the style guidelines of this project
- [ ] 👀 I have performed a self-review of my own code
- [ ] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [ ] ⚠️ My changes generate no new warnings
- [ ] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [ ] ⬆️ I increased versions for any altered app according to semantic versioning

**➕ App addition**

If this PR is an app addition please make sure you have done the following.

- [x] 🪞 I have opened a PR on [truecharts/containers](https://github.com/truecharts/containers) adding the container to TrueCharts mirror repo.
- [x] 🖼️ I have added an icon in the Chart's root directory called `icon.png`

---

_Please don't blindly check all the boxes. Read them and only check those that apply.
Those checkboxes are there for the reviewer to see what is this all about and
the status of this PR with a quick glance._

### CODE REVIEW

File: library/common/templates/lib/util/_chartcontext.tpl [hunk 1]
```diff
@@ -56,10 +56,28 @@
...
+      {{- $traefikportalhook := lookup "v1" "ConfigMap" $namespace "portalhook" | default dict -}}
+      {{/* If there is no portalhook */}}
+      {{- if not $traefikportalhook -}}
+        {{/* Get all configmaps */}}
+        {{- $hooks := lookup "v1" "ConfigMap" "" "" -}}
+
+        {{- $portalHooks := list -}}
+        {{- range $hook := $hooks -}}
+          {{/* Filter portalhook-* */}}
+          {{- if startswith $hook.metadata.name "portalhook-" -}}
```
Comment (stavros-k) [inline library/common/templates/lib/util/_chartcontext.tpl:70]:
```suggestion
          {{- if hasPrefix $hook.metadata.name "portalhook-" -}}
```

**Comment by Ornias1993** [comment]
@stavros-k Can this be merged?

**Comment by stavros-k** [comment]
> @stavros-k Can this be merged?

I'm not sure, i'll need to recheck it a bit after we changed the portal hook naming.

File: library/common/templates/lib/util/_chartcontext.tpl [hunk 2]
```diff
@@ -59,7 +59,30 @@
...
+            {{- end -}}
+          {{- end -}}
+        {{- end -}}
+
+        {{/* If there is only one portalhook, use it */}}
+        {{- if eq (len $portalHooks) 1 -}}
+          {{- $traefikportalhook = index $portalHooks 0 -}}
+        {{- end -}}
+        {{/* else??? */}}
+      {{- end -}}
```
Comment (Ornias1993) [inline library/common/templates/lib/util/_chartcontext.tpl:84]:
```suggestion
        {{/* use the first available portalhook */}}
        {{- $traefikportalhook = index $portalHooks 0 -}}
      {{- end -}}
```

**Review by Ornias1993 — APPROVED** [review]

**Comment by Ornias1993** [comment]
@stavros-k I've added another edge case.
It should now have less chance of defaulting to loadbalancer while on ingress.

File: library/common/Chart.yaml [hunk 3]
```diff
@@ -15,4 +15,4 @@ maintainers:
 name: common
 sources: null
 type: library
-version: 14.3.0
+version: 14.3.1
```
Comment (Ornias1993) [inline library/common/Chart.yaml:18]:
```suggestion
version: 14.3.2
```

**Pull request merged**